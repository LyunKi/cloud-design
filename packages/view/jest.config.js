module.exports = {
  ...require('../../jest.config.base'),
  preset: 'react-native',
  testEnvironment: 'jsdom',
  rootDir: '../..',
  roots: [`<rootDir>/packages/view`],
  testMatch: ['<rootDir>/packages/view/test/**/*.[jt]s?(x)'],
}
