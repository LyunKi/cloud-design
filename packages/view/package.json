{
  "name": "@cloud-design/view",
  "description": "Cloud design UI lib View component",
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "files": [
    "dist"
  ],
  "version": "0.1.0",
  "scripts": {
    "build": "tsc",
    "test": "jest",
    "test:watch": "jest --watch ",
    "test:updateSnap": "jest --updateSnapshot",
    "prepublish": "pnpm build"
  },
  "devDependencies": {
    "@cloud-design/configs": "workspace:*",
    "@types/react": "^18.0.15",
    "react": "^18.2.0",
    "react-native": "^0.69.4"
  },
  "peerDependencies": {
    "@cloud-design/configs": "workspace:*",
    "react": "^18.2.0",
    "react-native": "^0.69.4"
  }
}
