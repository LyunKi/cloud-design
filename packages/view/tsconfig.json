{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@cloud-design/view": ["./src"]
    }
  }
}
