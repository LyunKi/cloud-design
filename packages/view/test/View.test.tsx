import { render } from '@testing-library/react-native'
import { View } from '@cloud-design/view'
import React from 'react'

describe('<View />', () => {
  it('render correctly', () => {
    const snapshot = render(<View />).toJSON()
    expect(snapshot).toMatchSnapshot()
  })
})
