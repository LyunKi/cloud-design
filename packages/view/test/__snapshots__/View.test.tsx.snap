// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<View /> render correctly 1`] = `<View />`;
